client/lobby/CBonusSelection.cpp

```diff
@@ -553,41 +553,36 @@ CBonusSelection::CRegion::CRegion(CBonusSelection * _owner, bool _accessible, bo
 
 	std::string prefix = campDsc.campPrefix + desc.infix + "_";
 	std::string suffix = colors[campDsc.colorSuffixLength - 1][owner->getCampaign()->camp->scenarios[myNumber].regionColor];
-	static const std::string infix[] = {"En", "Se", "Co"};
-	for(int g = 0; g < ARRAY_COUNT(infix); g++)
-	{
-		graphics[g] = BitmapHandler::loadBitmap(prefix + infix[g] + suffix + ".BMP");
-	}
-	pos.w = graphics[0]->w;
-	pos.h = graphics[0]->h;
-
-}
+	graphicsNotSelected = std::make_shared<CPicture>(prefix + "En" + suffix + ".BMP");
+	graphicsNotSelected->disable();
+	graphicsSelected = std::make_shared<CPicture>(prefix + "Se" + suffix + ".BMP");
+	graphicsSelected->disable();
+	graphicsStriped = std::make_shared<CPicture>(prefix + "Co" + suffix + ".BMP");
+	graphicsStriped->disable();
+	pos.w = graphicsNotSelected->bg->w;
+	pos.h = graphicsNotSelected->bg->h;
 
-CBonusSelection::CRegion::~CRegion()
-{
-	for(auto & elem : graphics)
-	{
-		SDL_FreeSurface(elem);
-	}
 }
 
 void CBonusSelection::CRegion::show(SDL_Surface * to)
 {
-	//const SCampPositions::SRegionDesc & desc = owner->campDescriptions[owner->ourCampaign->camp->header.mapVersion].regions[myNumber];
 	if(!accessible)
 	{
-		// Show as striped
-		blitAtLoc(graphics[2], 0, 0, to);
+		graphicsNotSelected->disable();
+		graphicsSelected->disable();
+		graphicsStriped->enable();
```

Comment:
You should not call enable/disable during rendering. It should be done on condition change. It that is too difficult, do not capture SHOWALL and call `child->show()` here

Reply:
I'm already in process of cleanup of ```show``` and ```showAll``` .
Just not quite finished it yet.